src/hdf/fractalhead.c

```diff
@@ -4,334 +4,342 @@
+  uint8_t typeandversion;
+  uint64_t unknown, heap_header_address, block_offset, block_size, offset,
+      length;
+  long store;
+  struct DIR *dir;
+  struct MYSOFA_ATTRIBUTE *attr;
+
+  UNUSED(offset);
+  UNUSED(block_size);
+  UNUSED(block_offset);
+
+  if (reader->recursive_counter >= 10)
+    return MYSOFA_INVALID_FORMAT;
+  else
+    reader->recursive_counter++;
+
+  /* read signature */
+  if (fread(buf, 1, 4, reader->fhd) != 4 || strncmp(buf, "FHDB", 4)) {
+    log("cannot read signature of fractal heap indirect block\n");
+    return MYSOFA_INVALID_FORMAT;
+  }
+  log("%08" PRIX64 " %.4s stack %d\n", (uint64_t)ftell(reader->fhd) - 4, buf,
+      reader->recursive_counter);
+
+  if (fgetc(reader->fhd) != 0) {
+    log("object FHDB must have version 0\n");
+    return MYSOFA_UNSUPPORTED_FORMAT;
+  }
+
+  /* ignore heap_header_address */
+  if (fseek(reader->fhd, reader->superblock.size_of_offsets, SEEK_CUR) < 0)
+    return errno;
+
+  size = (fractalheap->maximum_heap_size + 7) / 8;
+  block_offset = readValue(reader, size);
+
+  if (fractalheap->flags & 2)
+    if (fseek(reader->fhd, 4, SEEK_CUR))
+      return errno;
+
+  offset_size = ceilf(log2f(fractalheap->maximum_heap_size) / 8);
+  if (fractalheap->maximum_direct_block_size < fractalheap->maximum_size)
+    length_size = ceilf(log2f(fractalheap->maximum_direct_block_size) / 8);
+  else
+    length_size = ceilf(log2f(fractalheap->maximum_size) / 8);
+
+  log(" %d %" PRIu64 " %d\n", size, block_offset, offset_size);
+
+  /*
+   * 00003e00  00 46 48 44 42 00 40 02  00 00 00 00 00 00 00 00
+   |.FHDB.@.........| 00003e10  00 00 00 83 8d ac f6 >03  00 0c 00 08 00 04 00
+   00  |................| 00003e20  43 6f 6e 76 65 6e 74 69  6f 6e 73 00 13 00
+   00 00  |Conventions.....| 00003e30  04 00 00 00 02 00 00 00  53 4f 46 41< 03
+   00 08 00  |........SOFA....| 00003e40  08 00 04 00 00 56 65 72  73 69 6f 6e
+   00 13 00 00  |.....Version....| 00003e50  00 03 00 00 00 02 00 00  00 30 2e
+   36 03 00 10 00  |.........0.6....| 00003e60  08 00 04 00 00 53 4f 46  41 43
+   6f 6e 76 65 6e 74  |.....SOFAConvent| 00003e70  69 6f 6e 73 00 13 00 00  00
+   13 00 00 00 02 00 00  |ions............| 00003e80  00 53 69 6d 70 6c 65 46 72
+   65 65 46 69 65 6c 64  |.SimpleFreeField| 00003e90  48 52 49 52 03 00 17 00 08
+   00 04 00 00 53 4f 46  |HRIR.........SOF| 00003ea0  41 43 6f 6e 76 65 6e 74 69
+   6f 6e 73 56 65 72 73  |AConventionsVers| 00003eb0  69 6f 6e 00 13 00 00 00 03
+   00 00 00 02 00 00 00  |ion.............|
+   *
+   */
+  do {
+    typeandversion = (uint8_t)fgetc(reader->fhd);
+    offset = readValue(reader, offset_size);
+    length = readValue(reader, length_size);
+    if (offset > 0x10000000 || length > 0x10000000)
+      return MYSOFA_UNSUPPORTED_FORMAT;
+
+    log(" %d %4" PRIX64 " %" PRIX64 " %08lX\n", typeandversion, offset, length,
+        ftell(reader->fhd));
+
+    /* TODO: for the following part, the specification is incomplete */
```

Comment:
Could the unspecified part be specified in the netcdf4 format ? Just a guess, I didn't find anything with quick googling